cmd/diff/cmddiff.go

```diff
@@ -4,12 +4,102 @@
 package diff
 
 import (
+	"fmt"
+	"os"
+
 	"github.com/spf13/cobra"
 	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/kubectl/pkg/cmd/apply"
 	"k8s.io/kubectl/pkg/cmd/diff"
 	"k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
 )
 
 func NewCmdDiff(f util.Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
-	return diff.NewCmdDiff(f, ioStreams)
+	options := diff.NewDiffOptions(ioStreams)
+	cmd := &cobra.Command{
+		Use:                   "diff DIRECTORY",
+		DisableFlagsInUseLine: true,
+		Short:                 i18n.T("Diff local config against cluster applied version"),
+		Args:                  cobra.MaximumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			util.CheckErr(Initialize(options, f, args))
+			util.CheckErr(options.Run())
+		},
+	}
+
+	return cmd
+}
+
+// Initialize fills in the DiffOptions in preparation for DiffOptions.Run().
+// Returns error if there is an error filling in the options or if there
+// is not one argument that is a directory.
+func Initialize(o *diff.DiffOptions, f util.Factory, args []string) error {
+	var err error
+
+	// Validate the only argument is a (package) directory path.
+	dir, err := parseDirArg(args)
+	if err != nil {
+		return err
+	}
+	o.FilenameOptions.Filenames = []string{dir}
+	o.FilenameOptions.Recursive = true
+
+	o.OpenAPISchema, err = f.OpenAPISchema()
+	if err != nil {
+		return err
+	}
+
+	o.DiscoveryClient, err = f.ToDiscoveryClient()
+	if err != nil {
+		return err
+	}
+
+	o.DynamicClient, err = f.DynamicClient()
+	if err != nil {
+		return err
+	}
+
+	o.DryRunVerifier = &apply.DryRunVerifier{
+		Finder:        util.NewCRDFinder(util.CRDFromDynamic(o.DynamicClient)),
+		OpenAPIGetter: o.DiscoveryClient,
+	}
+
+	o.CmdNamespace, o.EnforceNamespace, err = f.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	o.Builder = f.NewBuilder()
+
+	// We don't support server-side apply diffing yet.
+	o.ServerSideApply = false
+	o.ForceConflicts = false
+
+	return nil
+}
+
+// parseDirArg returns the validated directory path
+// or an error if there are more than one arguments,
+// or the path is not a directory.
+func parseDirArg(args []string) (string, error) {
```

Comment:
This looks very similar to the code at https://github.com/kubernetes-sigs/cli-utils/blob/09643102a8fcfd633696a6547e29e44b45c117dc/pkg/apply/path.go#L28-L49

Could we avoid the duplication?

Reply:
Yes. I was able to reuse the same function as the `apply` and `destroy` command--`DemandOneDirectory`. I had to move `path.go` and its unit test to a common directory `pkg/common`. Have a look.